h3py/api/_api_template.py

```diff
@@ -0,0 +1,228 @@
+import h3py.h3core as h3core
+import h3py.util as u
+
+# this module tries to DRY-up API code which is repeated across
+# modules. Not sure if this function closure is the best solution.
+# There doesn't seem to be any obvious best-practice for
+# programattically/dynamically creating modules
+
+# another approach: we could also just use `exec()`
+
+
+def _api_functions(
+    _in_scalar,
+    _out_scalar,
+    _in_collection,
+    _out_collection,
+):
+    def hex2int(h):
+        return u.hex2int(h)
+
+    def int2hex(x):
+        return u.int2hex(x)
+
+    def num_hexagons(resolution):
+        return h3core.num_hexagons(resolution)
+
+    def hex_area(resolution, unit='km'):
+        return h3core.hex_area(resolution, unit)
+
+    def edge_length(resolution, unit='km'):
+        return h3core.edge_length(resolution, unit)
+
+    def is_cell(h):
```

Comment:
Do you want to name this `h3_is_valid` to be consistent with the C API?

Reply:
See my comments below, but I was worried that `h3_is_valid` might not be clear to users that it **does not** validate an edge, but that it **does** validate hexagons and pentagons. I was trying out `cell` experimentally. Overall, names are cosmetic and I'm happy to move them around. I just wanted to take a chance to try out different names in the hopes of making the API easier to understand.

Reply:
It is called `h3_is_valid` in the default "old" API.

Reply:
I've updated the naming to match the current python implementation.